packages/catalog-process/src/services/catalogService.ts

```diff
@@ -1198,17 +1211,25 @@ export function updateDraftDescriptorLogic({
   );
 }
 
-export function publishDescriptorLogic({
+export async function publishDescriptorLogic({
   eserviceId,
   descriptorId,
   authData,
   eService,
+  listAgreements,
 }: {
   eserviceId: EServiceId;
   descriptorId: DescriptorId;
   authData: AuthData;
   eService: WithMetadata<EService> | undefined;
-}): CreateEvent<EServiceEvent> {
+  listAgreements: (
```

Comment:
possible improvement: in this logic we need just to know if at least one agreement exists.
We could add the pagination to this function, and use `offset = 0` and `limit = 1` to retrieve just one element, instead of returning all of them (which could potentially be several thousands)